### PR TITLE
Refactor/table style and behavior

### PR DESCRIPTION
**Pull request by samufacanha2**

### CODE REVIEW

File: src/components/MultiContractToolTip/index.tsx [hunk 1]
```diff
@@ -31,21 +26,18 @@ export const MultiContractToolTip: React.FC<IMultiContract> = ({
     return reducedContract;
   };
 
-  let msg = '';
-  Object.entries(reduceContracts()).forEach(([contrct, number]) => {
-    msg += `${ContractsIndex[contrct]}: ${number}x\n`;
-  });
+  const msg = Object.entries(reduceContracts())
+    .map(([contrct, number]) => {
```
Comment (marcioecom) [inline src/components/MultiContractToolTip/index.tsx:30]: typo `contrct`

File: src/components/TableV2/index.tsx [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+import { IPaginatedResponse, IRowSection } from '@/types/index';
+import { setQueryAndRouter } from '@/utils';
+import { useDidUpdateEffect } from '@/utils/hooks';
+import { processRowSectionsLayout } from '@/utils/table';
+import { useRouter } from 'next/router';
+import React, { useEffect, useState } from 'react';
+import { BsFillArrowUpCircleFill } from 'react-icons/bs';
+import { IoReloadSharp } from 'react-icons/io5';
+import { useQuery } from 'react-query';
+// import { VscJson } from 'react-icons/vsc';
```
Comment (marcioecom) [inline src/components/TableV2/index.tsx:12]: I think this comment could be removed

File: src/components/TableV2/index.tsx [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+    | 'votes'
+    | 'delegations'
+    | 'nfts'
+    | 'validatorsList'
+    | 'rewards'
+    | 'marketplaces'
+    | 'launchPad';
+
+  header: string[];
+  rowSections?: (item: any) => IRowSection[] | undefined;
```
Comment (marcioecom) [inline src/components/TableV2/index.tsx:63]: I think the question mark operator is same that `| undefined`

File: src/components/TableV2/index.tsx [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    try {
+      const response = await request(page, limit);
+      if (!response.error && dataName) {
+        responseFormatted = {
+          items: response.data[dataName],
+          totalPages: response?.pagination?.totalPages,
+        };
+        return responseFormatted;
+      }
+      // setPage(1);
```
Comment (marcioecom) [inline src/components/TableV2/index.tsx:112]: should this be here?

File: src/components/TableV2/index.tsx [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+      return [];
+    } catch (error) {
+      console.error(error);
+    }
```
Comment (marcioecom) [inline src/components/TableV2/index.tsx:116]: In others places we call toast also, here shouldn't?

File: src/utils/contracts/index.ts [hunk 6]
```diff
@@ -467,7 +478,7 @@ export enum contractTableHeaders {
   'BLS public key',
   'Public Key',
   'Bucket Id',
-  'Asset Id',
+  // 'Asset Id',
```
Comment (marcioecom) [inline src/utils/contracts/index.ts:481]: should this comment be here?

File: src/components/TableV2/index.tsx (hunk 4 above)
Comment (lucasportella) [inline src/components/TableV2/index.tsx:112]: if error current page should not change to first page? if so, delete comment

File: src/utils/contracts/index.ts (hunk 6 above)
Comment (lucasportella) [inline src/utils/contracts/index.ts:481]: remove comment

File: src/pages/transactions/index.tsx [hunk 7]
```diff
@@ -96,6 +94,15 @@ export const requestTransactionsDefault = async (
           ) {
             assets.push(contract.parameter.currencyID);
           }
+
```
Comment (lucasportella) [inline src/pages/transactions/index.tsx:97]: the loop is doing too much things, should be abstracted to smaller functions handling each case individually

File: src/pages/transactions/index.tsx [hunk 8]
```diff
@@ -119,6 +126,15 @@ export const requestTransactionsDefault = async (
             transaction.precision =
               assetPrecisions[contract.parameter.currencyID];
           }
+          if (contract?.type === TransactionType.Claim) {
```
Comment (lucasportella) [inline src/pages/transactions/index.tsx:129]: same here, also, is it necessary to call findReceipt with same args 2 times in the same page?

**Review by lucasportella — CHANGES_REQUESTED** [review]

File: src/components/TableV2/index.tsx (hunk 5 above)
Comment (samufacanha2) [reply src/components/TableV2/index.tsx:116]: if it catches an error, it will render an empty table. In an empty table, the user can refetch

**Review by lucasportella — APPROVED** [review]

**Pull request merged**